### PR TITLE
Move RTD files

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

These were smoothly moved into `docs` in a previous PR

This completes their removal with the RTD.org side of things now pointing to the `docs` folder

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@stephprince This should be a quick one

**Comment by CodyCBakerPhD** [comment]
(The external link check fail is the usual ontobee downtime)

(I don't think this deserves changelog)

File: pyproject.toml [hunk 1]
```diff
@@ -45,7 +45,8 @@ dependencies = [
     "click",
     "tqdm",
     "isodate",
-    "numpy>=1.22.0,<2.0.0" # TODO: add compatibility for 2.0
+    "numpy>=1.22.0,<2.0.0", # TODO: add compatibility for 2.0 - HDMF Zarr and others also still have some troubles
+    "h5py<3.12.0", # TODO: remove pin when https://github.com/h5py/h5py/issues/2505 is fixed and released
```
Comment (CodyCBakerPhD) [inline pyproject.toml:49]: Aaaand cool - this just happened 2 hours ago 🙃

**Review by stephprince — APPROVED** [review]
looks good to me!

**Pull request merged**